engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -62,13 +69,23 @@ class EngineJobExecutionActor(replyTo: ActorRef,
   // When Pending, the FSM always has NoData
   when(Pending) {
     case Event(Execute, NoData) =>
+      requestExecutionToken()
+      goto(RequestingExecutionToken)
+  }
+
+  when(RequestingExecutionToken) {
+    case Event(JobExecutionTokenDispensed(jobExecutionToken), NoData) =>
+      executionToken = Option(jobExecutionToken)
       if (restarting) {
         val jobStoreKey = jobDescriptorKey.toJobStoreKey(workflowId)
         jobStoreActor ! QueryJobCompletion(jobStoreKey, jobDescriptorKey.call.task.outputs)
         goto(CheckingJobStore)
       } else {
         prepareJob()
       }
+    case Event(JobExecutionTokenDenied(positionInQueue), NoData) =>
+      log.debug("Token denied so cannot start yet. Currently position {} in the queue", positionInQueue)
+      stay()
```

Comment:
Oh right I forgot they were queued up so that they would eventually get one without having to ask